.github/workflows/main.yml

```diff
@@ -0,0 +1,52 @@
+name: Build
+on:
+  push:
+    branches:
+      - staging
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: '0'
+
+      - name: Setup flutter
+        uses: subosito/flutter-action@v1
+        with:
+          channel: stable
+
+      - name: Get dependencies
+        run: flutter pub get
+
+      - name: Run build runner
+        run: flutter pub run build_runner build --delete-conflicting-outputs
+
+      - name: Build Website
+        run: flutter build web -t lib/main.widgetbook.dart
+
+      - name: Install Widgetbook CLI
+        run: dart pub global activate widgetbook_cli
+
+      - name: Upload Widgetbook
+        run: |
+          widgetbook \
+                --path "./" --api-key ${{ secrets.WIDGETBOOK_CLOUD_API_KEY }} \
+                --branch \
+                $GITHUB_HEAD_REF \
+                --repository \
+                $GITHUB_REPOSITORY \
+                --commit \
+                $GITHUB_SHA \
+                --actor \
+                $GITHUB_ACTOR \
+                --base-branch \
+                $GITHUB_BASE_REF \
+                --base-commit \
+                ${{ github.event.pull_request.base.sha }} \
+                --pr ${{ github.event.number }} \
+                --github-token ${{ secrets.GITHUB_TOKEN }} \
+                --git-provider \
+                "GitHub"
```

Comment:
This won't compile for now since we don't have `build_runner` and `widgetbook_generator` set up

Reply:
I think we need `widgetbook` CLI with the following arguments:
- `actor`
- `repository`
- `api-key`

Reply:
Would this work? 82e0b6f167ba8161c82414e657112363fc985f22

Reply:
That works, but you can run the whole workflow in the `app_widgetbook` folder instead of specifying it for each step

Reply:
Oops sorry didn't understand that earlier 😅
Changed it 👍